File: creational/factory/python/simple_factory.py
from product import Logger
from concrete_product import ConsoleLogger
from concrete_product import FileLogger


def create_logger(logger_type) -> Logger:
    logger_types = {
        "console": ConsoleLogger,
        "file": FileLogger,
    }

    if logger_type not in logger_types:
        raise ValueError("Invalid logger type provided")

    return logger_types[logger_type]()

File: creational/factory/typescript/simple-factory.ts
import { ConsoleLogger, FileLogger } from './concrete-product';

export const simpleFactory = (loggerType: string) => {
	const loggerTypes = {
		console: ConsoleLogger,
		file: FileLogger,
	};

	if (!(loggerType in loggerTypes)) throw new Error('Invalid logger type.');

	return new loggerTypes[loggerType as keyof typeof loggerTypes]();
};
